### ISSUE TITLE
feat(flows): RBIG warm-start → MLE fine-tune hybrid (rbig_to_trainable)

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/gauss_flows#127]
## Problem / Request

Provide a hybrid that uses `fit_rbig` for a closed-form warm start, transfers the fit into a trainable `gaussianization_flow`, then fine-tunes with `fit_gaussianization_flow` — realizing the Gaussianization Flows "easier initialization" claim.

## User Story

> As a user training Gaussianization flows, I want a closed-form warm start transferred into the trainable parameterization, so that training converges faster and generalizes better on small datasets.

## Motivation

The RBIG package stops at the closed-form fit; this issue is the bridge that turns that fit into a differentiable, fine-tunable flow. It is low-risk, high-leverage, and de-risks the estimator and benchmark issues by giving every flow a good warm start.

## Proposed API

New file `src/gauss_flows/_src/flows/rbig_init.py`:

```python
def rbig_to_trainable(rbig_flow, *, rotation="householder") -> Transformed:
    """Copy GMM params; convert fixed PCA rotations to learnable rotations."""

def fit_rbig_then_finetune(
    key, x, *, n_layers=8, n_components=8, max_epochs=100,
) -> tuple[Transformed, dict]
```

## Mathematical Notes

```text
Fine-tune objective:  L(θ) = D_KL[p_z(G_θ(x)) ‖ N(0, I)] = −E[log p_model(x)]   (MLE)
initialized at the RBIG solution.
```

Parameter transfer between the two parameterizations:

- `fit_rbig` builds `[FixedRotation.from_data (PCA), MixtureGaussianCDF (sklearn EM)]` blocks; `gaussianization_flow` builds `[MixtureGaussianCDF, HouseholderRotation]` blocks.
- (a) HistogramCDF/empirical marginals → parametric `MixtureGaussianCDF`/`MixtureLogisticCDF`: fit the mixture to the RBIG per-dim CDF (match quantiles), **not** raw data, to preserve the warm start.
- (b) fixed PCA `FixedRotation` → learnable `HouseholderRotation`/`OrthogonalRotation`.

**Landmines:**
- Householder cannot represent an arbitrary orthogonal matrix with fewer than d reflections — use `n_reflections=n_dims` to match a PCA rotation, or project the PCA matrix onto a Householder product (watch the reflection-vs-rotation determinant sign).
- Verify the warm-started flow's initial NLL ≈ RBIG NLL before any gradient step (regression guard).

## References & Existing Code

- Meng, Song, Song & Ermon, *Gaussianization Flows*, AISTATS 2020, PMLR 108:4336–4345 ([arXiv:2003.01941](https://arxiv.org/abs/2003.01941)) — "easier to initialize … generalize better on small training sets"
- Existing: `fit_rbig`, `gaussianization_flow`, `fit_gaussianization_flow` in `src/gauss_flows/_src/flows.py` / `train.py`
- Constraint: `fit_rbig_coupling` requires an even d ≥ 2 — any hybrid path exercising the coupling RBIG variant must respect that

## Implementation Steps

- [ ] Add `rbig_to_trainable` in `src/gauss_flows/_src/flows/rbig_init.py`
- [ ] Add `fit_rbig_then_finetune` in the same module
- [ ] Re-export both via `src/gauss_flows/__init__.py`
- [ ] Regression guard: warm-started initial NLL ≈ RBIG NLL

## Acceptance Criteria (numeric)

- [ ] Warm-started initial NLL within 1% of `fit_rbig` NLL on the same data (transfer fidelity)
- [ ] After fine-tuning, NLL ≤ warm-start NLL and ≤ NLL of a random-init flow trained for the same epochs
- [ ] Small-data regime (n ≤ 500, d = 8): hybrid beats random-init flow test NLL
- [ ] Determinism under fixed key

## Definition of Done

- [ ] Code lands at the intended paths
- [ ] Public API exported via `src/gauss_flows/__init__.py`
- [ ] Tests pass: `make test`
- [ ] Lint + typecheck pass: `make lint && make typecheck`
- [ ] Docstrings (Google-style) on all public symbols

## Testing

- [ ] Unit test: param-copy exactness
- [ ] Analytic test: Householder reconstruction of a known rotation to 1e-4
- [ ] Statistical test: small-data test-NLL over seeds (mark `slow`)
- [ ] Property test: idempotent transfer

## Documentation

- [ ] API reference section; recipe "warm-starting a Gaussianization flow from RBIG"

## Out of Scope

New marginal layer types.

## Relationships

- Parent (theme epic): #123
- Blocked by: —
- Related: #130 (SINF max-K-SWD rotations — interacts with the warm start)
